hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsFastpathSession.java

```diff
@@ -209,31 +225,44 @@ private boolean fetchFastpathFileHandle() {
     return false;
   }
 
+  @VisibleForTesting
+  protected AbfsRestOperation executeFetchFastpathSessionToken()
+      throws AzureBlobFileSystemException {
+    return client.getReadFastpathSessionToken(path, eTag, tracingContext);
+  }
+
   @VisibleForTesting
   protected AbfsRestOperation executeFastpathOpen()
       throws AzureBlobFileSystemException {
     return client.fastPathOpen(path, eTag, fastpathSessionInfo, tracingContext);
   }
 
   @VisibleForTesting
-  protected AbfsRestOperation executeFetchFastpathSessionToken()
+  protected AbfsRestOperation executeFastpathClose()
       throws AzureBlobFileSystemException {
-    return client.getReadFastpathSessionToken(path, eTag, tracingContext);
+    return client.fastPathClose(path, eTag, fastpathSessionInfo,
+        tracingContext);
   }
 
   @VisibleForTesting
   int getSessionRefreshIntervalInSec() {
-    return sessionRefreshIntervalInSec;
+    rwLock.readLock().lock();
+    try {
+      return sessionRefreshIntervalInSec;
+    } finally {
+      rwLock.readLock().unlock();
+    }
   }
 
   @VisibleForTesting
   void setConnectionMode(AbfsConnectionMode connMode) {
-    this.fastpathSessionInfo.setConnectionMode(connMode);
+    updateConnectionMode(connMode);
   }
 
   @VisibleForTesting
   void setAbfsFastpathSessionInfo(AbfsFastpathSessionInfo sessionInfo) {
-    this.fastpathSessionInfo = sessionInfo;
+    updateAbfsFastpathSessionToken(sessionInfo.getSessionToken(), sessionInfo.getSessionTokenExpiry());
```

Comment:
this is not atomic - each individual method is atomic but not all. simplest to take the write lock and assign the passed in sessionInfo.

Reply:
This method is removed